requests.go

```diff
@@ -92,16 +92,11 @@ func (c *Client) req(method, path string, body io.Reader, intercept func(*http.R
 }
 
 func (c *Client) mkcol(path string) int {
-	rs, err := c.req("MKCOL", path, nil, nil)
-	if err != nil {
+	rs, _ := c.req("MKCOL", path, nil, nil)
+	if rs == nil {
 		return 400
 	}
 	defer rs.Body.Close()
-
-	if rs.StatusCode == 201 || rs.StatusCode == 405 {
-		return 201
-	}
```

Comment:
Not too sure about this one tbh. Seems like it was there for a reason. Do you know why?

Reply:
Yeah, don't really know either why `405 Method not allowed` is converted to 201 - especially because the package should have full control over which HTTP method is used - but I am also not very experienced with the WebDAV protocol.

Reply:
you'll get a 405 if the collection already exists.